Documentation/botr/readytorun-format.md

```diff
@@ -0,0 +1,494 @@
+
+    // Long mul/div/shift ops
+    READYTORUN_HELPER_LMul                      = 0xC0,
+    READYTORUN_HELPER_LMulOfv                   = 0xC1,
+    READYTORUN_HELPER_ULMulOvf                  = 0xC2,
+    READYTORUN_HELPER_LDiv                      = 0xC3,
+    READYTORUN_HELPER_LMod                      = 0xC4,
+    READYTORUN_HELPER_ULDiv                     = 0xC5,
+    READYTORUN_HELPER_ULMod                     = 0xC6,
+    READYTORUN_HELPER_LLsh                      = 0xC7,
+    READYTORUN_HELPER_LRsh                      = 0xC8,
+    READYTORUN_HELPER_LRsz                      = 0xC9,
+    READYTORUN_HELPER_Lng2Dbl                   = 0xCA,
+    READYTORUN_HELPER_ULng2Dbl                  = 0xCB,
+
+    // 32-bit division helpers
+    READYTORUN_HELPER_Div                       = 0xCC,
+    READYTORUN_HELPER_Mod                       = 0xCD,
+    READYTORUN_HELPER_UDiv                      = 0xCE,
+    READYTORUN_HELPER_UMod                      = 0xCF,
+
+    // Floating point conversions
+    READYTORUN_HELPER_Dbl2Int                   = 0xD0,
+    READYTORUN_HELPER_Dbl2IntOvf                = 0xD1,
+    READYTORUN_HELPER_Dbl2Lng                   = 0xD2,
+    READYTORUN_HELPER_Dbl2LngOvf                = 0xD3,
+    READYTORUN_HELPER_Dbl2UInt                  = 0xD4,
+    READYTORUN_HELPER_Dbl2UIntOvf               = 0xD5,
+    READYTORUN_HELPER_Dbl2ULng                  = 0xD6,
+    READYTORUN_HELPER_Dbl2ULngOvf               = 0xD7,
+
+    // Floating point ops
+    READYTORUN_HELPER_DblRem                    = 0xE0,
+    READYTORUN_HELPER_FltRem                    = 0xE1,
+    READYTORUN_HELPER_DblRound                  = 0xE2,
+    READYTORUN_HELPER_FltRound                  = 0xE3,
+
+#ifndef _TARGET_X86_
+    // Personality rountines
+    READYTORUN_HELPER_PersonalityRoutine        = 0xF0,
+    READYTORUN_HELPER_PersonalityRoutineFilterFunclet = 0xF1,
+#endif
+
+    //
+    // Deprecated/legacy
+    //
+
+    // JIT32 x86-specific write barriers
+    READYTORUN_HELPER_WriteBarrier_EAX          = 0x100,
+    READYTORUN_HELPER_WriteBarrier_EBX          = 0x101,
+    READYTORUN_HELPER_WriteBarrier_ECX          = 0x102,
+    READYTORUN_HELPER_WriteBarrier_ESI          = 0x103,
+    READYTORUN_HELPER_WriteBarrier_EDI          = 0x104,
+    READYTORUN_HELPER_WriteBarrier_EBP          = 0x105,
+    READYTORUN_HELPER_CheckedWriteBarrier_EAX   = 0x106,
+    READYTORUN_HELPER_CheckedWriteBarrier_EBX   = 0x107,
+    READYTORUN_HELPER_CheckedWriteBarrier_ECX   = 0x108,
+    READYTORUN_HELPER_CheckedWriteBarrier_ESI   = 0x109,
+    READYTORUN_HELPER_CheckedWriteBarrier_EDI   = 0x10A,
+    READYTORUN_HELPER_CheckedWriteBarrier_EBP   = 0x10B,
+
+    // JIT32 x86-specific exception handling
+    READYTORUN_HELPER_EndCatch                  = 0x110,
+};
+```
+
+# References
+
+[1: Managed Executables with Native Code](https://microsoft.sharepoint.com/teams/netfx/CLR/coreclr/Shared%20Documents1/ReadyToRun/ManagedNativeCode.docx)
```

Comment:
This is an internal-only link.

Reply:
Thanks for pointing that out, removed in 2nd commit.